web/packages/teleterm/src/ui/services/clusters/clustersService.ts

```diff
@@ -640,3 +640,34 @@ export function makeApp(source: tsh.App): App {
 
   return { ...source, addrWithProtocol };
 }
+
+/** Returns a URL that can be used to open the app in a browser. */
+export function getWebAppLaunchUrl(
```

Comment:
It seems to me that having small, specialized functions is better than adding more properties to `makeApp` (that we do above). We should try to get rid of this function.